### ISSUE TITLE
cat << EOF | cat << end

### ISSUE THREAD

**Issue by Chr-ss** [Chr-ss/minishell#115]
Dub error

**Comment by Chr-ss** [comment]
![IMG_20240802_130701](https://github.com/user-attachments/assets/1696eabd-0665-4b7f-8e69-e58555cc6334)

**Comment by Chr-ss** [comment]
![IMG_20240802_131451](https://github.com/user-attachments/assets/de3c18f9-2262-4b75-b4ba-99d521104d7d)


Same issue for append

**Comment by Chr-ss** [comment]
Potential idea:

If the input is not the read end of the pipe close the piperead. 

ls is reading from heredoc we don't need the pipe. 

```
void	ft_dup_pipes(t_stream *iostream)
{
	if (iostream->input != -1)
	{
		if (dup2(iostream->input, STDIN_FILENO) == -1)
			exit(errno);
	}
	if (iostream->input != iostream->pipes->curr_read)
		close(iostream->pipes->curr_read);
	if (iostream->output != -1)
	{
		if (dup2(iostream->output, STDOUT_FILENO) == -1)
			exit(errno);
	}
	if (iostream->output != iostream->pipes->curr_write)
		close(iostream->pipes->curr_write);
}
```

**Comment by Chr-ss** [comment]
https://github.com/rubenfin/minishell/blob/main/src/executing/dup.c

**Comment by Chr-ss** [comment]
fixed in 43345be7f15be735accb2946b37a11942065280a

@sairutra Please look at the changes made.
